Sources_v3/StreamChatUI/MessageActionsPopup/ChatMessageActionItem.swift

```diff
@@ -5,103 +5,37 @@
 import UIKit
 
 public struct ChatMessageActionItem {
-    public let title: String
-    public let icon: UIImage
-    public let isDestructive: Bool
+    public let name: Name
     public let action: () -> Void
 
     public init(
-        title: String,
-        icon: UIImage,
-        isDestructive: Bool,
+        name: Name,
         action: @escaping () -> Void
     ) {
-        self.title = title
-        self.icon = icon
-        self.isDestructive = isDestructive
+        self.name = name
         self.action = action
     }
 }
 
-public extension ChatMessageActionItem {
-    static func inlineReply(action: @escaping () -> Void) -> Self {
-        .init(
-            title: L10n.Message.Actions.inlineReply,
-            icon: UIImage(named: "icn_inline_reply", in: .streamChatUI)!,
-            isDestructive: false,
-            action: action
-        )
-    }
-
-    static func threadReply(action: @escaping () -> Void) -> Self {
-        .init(
-            title: L10n.Message.Actions.threadReply,
-            icon: UIImage(named: "icn_thread_reply", in: .streamChatUI)!,
-            isDestructive: false,
-            action: action
-        )
-    }
-
-    static func edit(action: @escaping () -> Void) -> Self {
-        .init(
-            title: L10n.Message.Actions.edit,
-            icon: UIImage(named: "icn_edit", in: .streamChatUI)!,
-            isDestructive: false,
-            action: action
-        )
-    }
-
-    static func copy(action: @escaping () -> Void) -> Self {
-        .init(
-            title: L10n.Message.Actions.copy,
-            icon: UIImage(named: "icn_copy", in: .streamChatUI)!,
-            isDestructive: false,
-            action: action
-        )
-    }
+extension ChatMessageActionItem {
+    public struct Name: RawRepresentable, Hashable {
+        public let rawValue: String
```

Comment:
What's the reason for having a wrapper type around `String`?

Reply:
Long time ago we had a discussing about extending the built-in type. The output was to avoid it if possible. That's the reason why the string representable type is introduced.

Reply:
fixed